src/test/java/com/waverleysoftware/BaseTest.java

```diff
@@ -1,16 +1,39 @@
 package com.waverleysoftware;
 
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.SoftAssertions;
+import org.testng.ITestContext;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
 
+import java.lang.reflect.Method;
+
+/**
+ * A simple base class for processing tests.
+ */
+@Slf4j
 public class BaseTest {
+
+    protected static final SoftAssertions SOFTASSERT = new SoftAssertions();
```

Comment:
according to Java conventions, each word in a constant should be splitted with `_` character